### ISSUE TITLE
yang list accessed by key and position too

### ISSUE THREAD

**Issue by johnkaramp** [CESNET/libyang#2328]
Hello,
I want to have a order-by user list (actually create the entries in a specific sorted order) with a key (string) that I can also access elements by positional index too, except from key.
I am aware that I can get list instance with
`/module-name:container/list[key1='1'][key2='2']`
and also get a specific instance from a list without keys
`/module-name:container/list-without-keys[3]`
However can I use the second to access the third element lets say of a list with keys?
If not, I suppose that i can use the standard way of moving elements where I want in a list with keys and ordered-by user, and then with the proper libyang api that I can iterate the list, if I want the first element (which is the smaller one in this case), it will be available in the first iteration already.
Thank you in advance.

**Comment by michalvasko** [comment]
Yes, you can use `/module-name:container/list-without-keys[3]` even for lists with keys but you have to use the full XPath, not Path. In practice, use `lyd_find_xpath()` and not `lyd_find_path()`.

**Comment by johnkaramp** [comment]
thank you! one more question, if I want to use for example count() not in an expression in some index but like an operation to count number of elements, is there a dedicated function like `lyd_find_xpath()` that I can run this operation and return an int?
